### ISSUE TITLE
XML configuration file errors

### ISSUE THREAD

**Issue by Telavian** [Azure/service-fabric-issues#374]
While creating some services and modifying the ServiceManifest.xml files I accidentally left an invalid character in one of them. This of course caused the deploy to fail and I received a generic message "There is an error in XML document (13, 21)".

This message did give me a location however I had modified several files and for some reason I thought this pertained to a different file. While researching I thought my issue was related to #111 so I spent a bunch of wasted time trying to figure things out.

It would be much more helpful if the message included the file name or path.

"There is an error in XML document c:\xyz\abc.xml at (10, 10)".

**Comment by oanapl** [comment]
Thank you for reporting, @Telavian.

What version of Service Fabric are you using? And what is the command you are executing?

For SF version 5.7.186.9494, when I add an extra character, I get this error message when I check the application package:

```powershell
PS C:\> Test-ServiceFabricApplicationPackage -ApplicationPackagePath D:\ServiceFabric\WordCount
False
Test-ServiceFabricApplicationPackage : Exception while parsing XML file: C:\Users\oanapl\AppData\Local\Temp\TestApplicationPackage_5236233986871\fsfbr5of.xfn\WordCount\WordCountServicePkg\ServiceManifest.xml
FileName: C:\Users\oanapl\AppData\Local\Temp\TestApplicationPackage_5236233986871\fsfbr5of.xfn\WordCount\WordCountServicePkg\ServiceManifest.xml
At line:1 char:1
+ Test-ServiceFabricApplicationPackage -ApplicationPackagePath D:\Servi ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Test-ServiceFabricApplicationPackage], FabricImageBuilderValidationException
    + FullyQualifiedErrorId : TestApplicationPackageErrorId,Microsoft.ServiceFabric.Powershell.TestApplicationPackage
```
When I try to register, I get this error:
```powershell
PS C:\> Register-ServiceFabricApplicationType -ApplicationPathInImageStore wc2
Register-ServiceFabricApplicationType : Exception while parsing XML file: C:\SfDevCluster\Data\ImageBuilderProxy\AppType\WordCount\WordCountServicePkg\ServiceManifest.xml
FileName: C:\SfDevCluster\Data\ImageBuilderProxy\AppType\WordCount\WordCountServicePkg\ServiceManifest.xml --> InvalidOperationException: There is an error in XML document (16, 71). --> 
XmlSchemaValidationException: The element 'Endpoints' in namespace 'http://schemas.microsoft.com/2011/01/fabric' cannot contain text. List of possible elements expected: 'Endpoint' in namespace 
'http://schemas.microsoft.com/2011/01/fabric'.
At line:1 char:1
+ Register-ServiceFabricApplicationType -ApplicationPathInImageStore wc ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (Microsoft.Servi...usterConnection:ClusterConnection) [Register-ServiceFabricApplicationType], FabricException
    + FullyQualifiedErrorId : RegisterApplicationTypeErrorId,Microsoft.ServiceFabric.Powershell.RegisterApplicationType
```

The error message includes the file name. Can you also give me more info about what invalid character you introduced and where in the file was it?

**Comment by Telavian** [comment]
I am doing everything from Visual Studio 2015 so I am not sure about the SF version.
Maybe VS just doesn't report the error but powershell does.

I am testing with a 1 node cluster.
I have version 1.6.20170508.2 of Service Fabric Tools for Visual Studio
The system services inside SF have a version of 5.6.220.9494.

Attached is a totally generic stateless service that VS created.
The only change is a single incorrect extra character 'd' in ServiceManifest.xml.

The line was `<StatelessServiceType ServiceTypeName="StatelessTestServiceType" />`
now it is `<StatelessServiceType ServiceTypeName="StatelessTestServiceType" />d`

[SFTest.zip](https://github.com/Azure/service-fabric-issues/files/1189035/SFTest.zip)

**Comment by oanapl** [comment]
Yes, VS doesn't show the full error message that the underlying Powershell cmdlets expose. 
@MikkelHegn / @dbreshears, can you follow up on tooling please?

**Comment by dbreshears** [comment]
This has been fixed in the 1.6.1 version of the Microsoft.VisualStudio.Azure.Fabric.MSBuild Nuget package which ships as part of our latest tooling.  I see the filename output as part of Test-ServiceFabricApplicationPackage

When I publish the SFTest sample project to local cluster I get 

```
2>Started executing script 'Deploy-FabricApplication.ps1'.
2>. 'C:\temp\SFTest\SFTest\Scripts\Deploy-FabricApplication.ps1' -ApplicationPackagePath 'C:\temp\SFTest\SFTest\pkg\Debug' -PublishProfileFile 'C:\temp\SFTest\SFTest\PublishProfiles\Local.5Node.xml' -DeployOnly:$false -ApplicationParameter:@{} -UnregisterUnusedApplicationVersionsAfterUpgrade $false -OverrideUpgradeBehavior 'None' -OverwriteBehavior 'SameAppTypeAndVersion' -SkipPackageValidation:$false -ErrorAction Stop
2>Test-ServiceFabricApplicationPackage : Exception while parsing XML file: C:\Users\devinb\AppData\Local\Temp\TestApplica
2>tionPackage_3349552716125\svvls3t3.5hc\Debug**\StatelessTestServicePkg\ServiceManifest.xml**
2>FileName: C:\Users\devinb\AppData\Local\Temp\TestApplicationPackage_3349552716125\svvls3t3.5hc\Debug\StatelessTestServi
2>cePkg\ServiceManifest.xml

```
